sdk/eventhub/azure-eventhub/tests/livetest/synctests/test_send.py

```diff
@@ -229,3 +231,190 @@ def test_send_batch_pid_pk(invalid_hostname, partition_id, partition_key):
+        assert event_data_batch.starting_published_sequence_number == 0
+        assert event_data.published_sequence_number == 0
+
+        message = receivers[0].receive_message_batch(timeout=10000)[0]
+        received = EventData._from_message(message)
+        assert received.body_as_str() == payload
+
+        event_data_list = [EventData(payload), EventData(payload)]
+        client.send_batch(event_data_list, partition_id="0")
+
+        partition_publishing_properties = client.get_partition_publishing_properties("0")
+        assert partition_publishing_properties["last_published_sequence_number"] == 2
+        assert event_data_list[0].published_sequence_number == 1
+        assert event_data_list[1].published_sequence_number == 2
+        message = receivers[0].receive_message_batch(timeout=10000)
+        assert len(message) == 2
+
+
+@pytest.mark.liveTest
+def test_idempotent_sender_partition_configs(connstr_receivers):
+    connection_str, receivers = connstr_receivers
+
+    client = EventHubProducerClient.from_connection_string(
+        connection_str,
+        enable_idempotent_partitions=True,
+    )
+
+    payload = "A1"
+    event_data_batch = client.create_batch(partition_id="0")
+    event_data = EventData(payload)
+    event_data_batch.add(event_data)
+    client.send_batch(event_data_batch)
+
+    partition_publishing_properties = client.get_partition_publishing_properties("0")
+    assert partition_publishing_properties["last_published_sequence_number"] == 0
+    assert partition_publishing_properties["producer_group_id"] == 0
+    assert partition_publishing_properties["owner_level"] == 0
+    assert event_data_batch.starting_published_sequence_number == 0
+    assert event_data.published_sequence_number == 0
+
+    partition_config_for_producer_2 = {
+        "0": PartitionPublishingConfiguration(
+            producer_group_id=partition_publishing_properties["producer_group_id"],
+            owner_level=partition_publishing_properties["owner_level"] + 1
+        )
+    }
+
+    client_2 = EventHubProducerClient.from_connection_string(
+        connection_str,
+        enable_idempotent_partitions=True,
+        partition_configs=partition_config_for_producer_2
+    )
+    event_data_list = [EventData(payload), EventData(payload)]
+    client_2.send_batch(event_data_list, partition_id="0")
+    partition_publishing_properties_2 = client_2.get_partition_publishing_properties("0")
+    # with the new owner_level, the sequence number starts from 0 again
+    assert partition_publishing_properties_2["last_published_sequence_number"] == 1
+    assert partition_publishing_properties_2["producer_group_id"] == 0
+    assert partition_publishing_properties_2["owner_level"] == partition_publishing_properties["owner_level"] + 1
+
+    # test producer-epoch-stolen
+    with pytest.raises(EventDataSendError):
+        # the first producer will get exploded as the link is stolen by the second producer with higher owner level
```

Comment:
probably dumb question, but is it possible for owner_levels to be the same ever? for example, if the user passes in the same level for both producers.

Reply:
yes, according to the service spec and my test, onwer_level could be set to the same value, it's the <producer-id, owner-level> pair identifies a producer.